Hotels/Models/HotelRoom.cs

```diff
@@ -0,0 +1,15 @@
+﻿namespace Hotels.Models
+{
+    public class HotelRoom
+    {
+        public long Id { get; set; }
+        public long HotelId { get; set; }
+        public long RoomId { get; set; }
+        public Hotel Hotel { get; set; }
+        public Room Room { get; set; }
+
+        public int RoomNumber { get; set; }
```

Comment:
I would probably put this at the top, next to `HotelId`, to reinforce that those are the PKs.